application/src/main/data/upgrade/3.6.3/schema_update.sql

```diff
@@ -14,6 +14,105 @@
 -- limitations under the License.
 --
 
+-- UPDATE PUBLIC CUSTOMERS START
+
+ALTER TABLE customer ADD COLUMN IF NOT EXISTS is_public boolean DEFAULT false;
+UPDATE customer SET is_public = true WHERE title = 'Public';
+
+-- UPDATE PUBLIC CUSTOMERS END
+
+-- UPDATE CUSTOMERS WITH SAME TITLE START
+
+CREATE OR REPLACE PROCEDURE update_customers_with_the_same_title()
```

Comment:
I suggest implementing this in Java since then we may add unit/integration tests. This is a critical code part where bugs are not acceptable. Too much logic in SQL.

Reply:
We may skip this only if @smatvienko-tb will accept this part.

Reply:
@smatvienko-tb, no needs to review this. I have talked to Andrew and we decided to move the upgrade script to Java.